internal/infrastructure/kubernetes/proxy/resource_provider.go

```diff
@@ -170,6 +170,17 @@ func (r *ResourceRender) Deployment() (*appsv1.Deployment, error) {
 		annotations = deploymentConfig.Pod.Annotations
 	}
 
+	// Get custom and selector labels for pod
```

Comment:
we should perform a `append` here, not replace

Reply:
+1, we should append some new labels and keep the default label

Reply:
if append the custom labels in this func:
https://github.com/envoyproxy/gateway/blob/773bdfbefc769f51779d5056547e88d12e1d9508/internal/infrastructure/kubernetes/proxy/resource_provider.go#L160
the custom labels will be tagged successfully on the pod:
https://github.com/envoyproxy/gateway/blob/773bdfbefc769f51779d5056547e88d12e1d9508/internal/infrastructure/kubernetes/proxy/resource_provider.go#L188-L190
but these custom labels will also be tagged on its deployment:
https://github.com/envoyproxy/gateway/blob/773bdfbefc769f51779d5056547e88d12e1d9508/internal/infrastructure/kubernetes/proxy/resource_provider.go#L178-L181

is this a reasonable behavior?

Reply:
probably call the func twice, and assign it to `deploymentLabels` and `podLabels` fields, and reuse them ?

Reply:
great idea